src/node.rs

```diff
@@ -244,10 +244,17 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
 
         let is_leader = self.raw_node.raft.state == raft::StateRole::Leader;
         if is_leader {
+            // Check if we stepped up as leader from follower state
+            if let Some(FollowerState::Committing(ref block_id)) = self.follower_state {
```

Comment:
This can probably also be combined with the block below. Basically if the leader state is none, you need to set the leader state and unset the follower state. So what you should end up with is `if self.leader_state.is_none() { if let Some(FollowerState::Committing(ref block_id)) = self.follower_state { set leader state to committing; } else { set leader state to building; init block; } set follower state to none; }`

Reply:
Yes. This could be combined. Updated the commit